### PR TITLE
Fixes #711: Adding ide:list-mine command.

### LINKED ISSUE #711: Add command to list all of "my" IDEs across all apps
As an Acquia Cloud user with many applications, I'd like a way to list all of my IDEs regardless of application.

### PR DESCRIPTION
**Pull request by grasmash**
**Motivation**

Fixes #711

**Proposed changes**

* Add a new `ide:list-mine` command.
* Rename `ide:list`	 to `ide:list-app`, leave an `ide:list` for backwards compatibility. 

**Alternatives considered**

Change behavior of `ide:list` to make application argument optional.

**Testing steps**

1. Follow the [contribution guide](https://github.com/acquia/cli/blob/master/CONTRIBUTING.md#building-and-testing) to set up your development environment.
2. Clear the kernel cache to pick up new and changed commands: `./bin/acli ckc`
3. Run `ide:list-app`
4. Run `ide:list-mine`

**Merge requirements**
- [x] _Bug_, _enhancement_, or _breaking change_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

**Comment by grasmash** [comment]
If we were going to model our command names off of Cloud API's patterns, we'd actually name the commands something like:

* account:ides
* application:ides
* ide:create
* ide:delete
* ide:info
* ide:open

File: src/Command/Ide/IdeListMineCommand.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ */
+class IdeListMineCommand extends IdeCommandBase {
+
+  protected static $defaultName = 'ide:list-mine';
+
+  /**
+   * {inheritdoc}.
+   */
+  protected function configure() {
+    $this->setDescription('List available Cloud IDEs belonging to you');
```
Comment (danepowell) [inline src/Command/Ide/IdeListMineCommand.php:22]:
```suggestion
    $this->setDescription('List Cloud IDEs belonging to you');
```

Normally when we talk about "available" Cloud IDEs, we're referring to available entitlements. Not sure if customers see it the same way, but could be confusing.

**Review by anavarre — APPROVED** [review]
```
./acli.phar ide:list-mine
 =======================================================================
  IDEs
 =======================================================================
  drupal7
  UUID: 4d287d5c-5e7d-4daa-b7c8-0c64c44618b1
  Application: MyD7App
  Subscription: MyD7App
  IDE URL: https://4d287d5c-5e7d-4daa-b7c8-0c64c44618b1.ide.ahdev.cloud
  Web URL: https://4d287d5c-5e7d-4daa-b7c8-0c64c44618b1.web.ahdev.cloud
 =======================================================================
  d8isnotdead
  UUID: c7c2134c-7192-4c50-800e-1c72b592edc7
  Application: Main App
  Subscription: Main App
  IDE URL: https://c7c2134c-7192-4c50-800e-1c72b592edc7.ide.ahdev.cloud
  Web URL: https://c7c2134c-7192-4c50-800e-1c72b592edc7.web.ahdev.cloud
 =======================================================================
  test
  UUID: abab4dd0-47d8-4db9-ad39-12ab33f35adf
  Application: Main App
  Subscription: Main App
  IDE URL: https://abab4dd0-47d8-4db9-ad39-12ab33f35adf.ide.ahdev.cloud
  Web URL: https://abab4dd0-47d8-4db9-ad39-12ab33f35adf.web.ahdev.cloud
 =======================================================================
 =======================================================================
```

File: src/Command/Ide/IdeListMineCommand.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  /**
+   * @param \Symfony\Component\Console\Input\InputInterface $input
+   * @param \Symfony\Component\Console\Output\OutputInterface $output
+   *
+   * @return int 0 if everything went fine, or an exit code
+   * @throws \Exception
+   */
+  protected function execute(InputInterface $input, OutputInterface $output) {
+    $acquia_cloud_client = $this->cloudApiClientService->getClient();
+    $account_ides = $acquia_cloud_client->request('get', '/account/ides');
```
Comment (danepowell) [inline src/Command/Ide/IdeListMineCommand.php:35]: Opened an upstream issue to add this endpoint: https://github.com/typhonius/acquia-php-sdk-v2/issues/208

Not a blocker for this PR, but feel free to weigh in there.

File: src/Command/Ide/IdeListMineCommand.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    $application_resource = new Applications($acquia_cloud_client);
+
+    if (count($account_ides)) {
+      $table = new Table($output);
+      $table->setStyle('borderless');
+      $table->setHeaders(['IDEs']);
+      foreach ($account_ides as $ide) {
+        $app_url_parts = explode('/', $ide->_links->application->href);
```
Comment (danepowell) [inline src/Command/Ide/IdeListMineCommand.php:43]: This looks like it should use the same output format as `ide:info`. Can we refactor this and `ide:info` to use a common method that accepts an array of `IDE` resources and pretty-prints a table?

Maybe even use it for `ide:list` as well, with a `verbose` parameter.

**Review by danepowell — COMMENTED** [review]
I was already working on a review when you merged 😬 I'll still leave these comments, maybe we can follow up.

File: src/Command/Ide/IdeListMineCommand.php (hunk 3 above)
Comment (anavarre) [reply src/Command/Ide/IdeListMineCommand.php:43]: Filed CLI-530 so your feedback is not lost.

**Pull request merged**